bluemira/base/reactor_config.py

```diff
@@ -234,21 +236,56 @@ def _check_args_are_strings(self, args: Iterable[str]):
             if not isinstance(a, str):
                 raise ReactorConfigError("args must be strings")
 
+    def _get_nested_filepaths(
+        self,
+        current_layer: str,
+        current_arg_key: str,
+        arg_keys: Tuple[str],
+        next_idx: int,
+    ) -> Dict:
+        if not Path(current_layer).resolve().is_file():
+            if isinstance(self._config_path, str):
+                current_layer = str(Path(self._config_path).parent / current_layer)
+            if not Path(current_layer).resolve().is_file():
+                raise FileNotFoundError(f"Cannot find file {current_layer}")
+
+        new_layer = self._read_json_file(current_layer)
+
+        # Dont need to load files repeatedly so put it back in config_data
+        if next_idx == 1:
+            self.config_data[current_arg_key] = new_layer
+        else:
+            tmp_dict = self.config_data
+            for key in arg_keys[: next_idx - 1]:
+                tmp_dict = tmp_dict[key]
+            tmp_dict[current_arg_key] = new_layer
+        return new_layer
```

Comment:
It would work but Oli suggested that storing the output would loose connection to the original file (eg in a notebook) which is a bit counter intuitive. Now we just read in the files every time they're accessed

Reply:
Wouldn't we still potentially need to load things recursively? For example, in this case, a user might expect `config.config_for("My Builder")` to unpack `opt_conditions`. 

```JSON
// config.json
{
  "My Builder": "$./mybuilder_config.json",
  ...
}
```

```JSON
// mybuilder_config.json
{
  "build_config": {
    "option_1": 1,
    "opt_conditions": "$./my_builder_opt_conditions.json"
   }
}
```

Or would this work and I'm misunderstanding?

Also, not entirely sure that I see the value in maintaining the file path link in code. The link is in the config file. Obviously, not a big issue, but I would be tempted to not worry about that link if it makes our lives easier.